nestedtensor/nested/monkey_patch.py

```diff
@@ -233,9 +233,14 @@ def new_fn(self):
     #     set_nt_method(function_name, getattr(functions, function_name))
 
     set_function(torch.conv2d, functions.conv2d)
-    set_function(torch.max_pool2d, functions.max_pool2d)
+    set_function(torch.nn.functional.max_pool2d, functions.max_pool2d)
```

Comment:
You should be overriding the "innermost" function and not the functional. So, "torch.max_pool2d" and not "torch.nn.functional.max_pool2d". I'm actually surprised this works via ```__torch_function__```.

Reply:
I see, it's due to the [torch function support for functionals](https://github.com/pytorch/pytorch/blob/master/torch/nn/functional.py#L2979). Eventually however we'll want to overwrite builtins to implement NestedTensor C++ kernels for natives functions and get JIT support.